crates/lang/src/reflect/trait_def/info.rs

```diff
@@ -152,3 +157,11 @@ pub trait TraitModulePath {
     /// This is just for convenience.
     const NAME: &'static str;
 }
+
+/// Implemented by call builders of smart contracts and by `TraitDefinitionRegistry`
+/// during codegen to apply the blanket implementation.
+///
+/// The blanket implementation is generated by `#[ink_lang::trait_definition]` and
+/// defined in the same place with the trait. It generates the body for methods of the trait.
+/// These methods contain cross-contract calls inside and define the `__ink_TraitInfo`.
```

Comment:
This has too many low level details about the implementation. I'd make the doc comments higher level and maybe write about the implementation details at a more relevant location

Reply:
The same idea here. It is a trait for codegen and it described how it works inside. If we will remove the blanket implementation we also will remove that trait.